### PR TITLE
zx - same image link

### LINKED ISSUE #2: Save Image Link
#### As a user I want to save a link to an image.

__Story__:

Now that we have an awesome landing page for our users to experience it is time
to introduce our first feature! In general, we want to deliver value to our
customers as fast as possible. As a result, we have decided that we want our
users to be able to save links to images, and immediately see that we have
saved the URL to the image.

One could argue that simply saving the image, and not showing it, provides some
value. Yes, you could split this story here. However, because saving image URLs
and viewing those images are both relatively small additions that work well
together, it is acceptable to complete them together. Note, on the other hand,
that we do not yet want to implement the image index. Its addition adds too
much complexity, so we’ll save that for a future story.

In general, feature addition, or stories, should be split into the smallest
deliverable chunk that provides value to your users with the caveat that it
might make sense to combine two, sometimes more, tiny and related features into
a single story.

__Acceptance criteria__:
- [ ] The landing page links to the image link submission form.
- [ ] The link is entered through a form.
- [ ] After the form is saved, the link is persisted in the database.
- [ ] After the form is submitted, the user is redirected to a page displaying
  the image.
- [ ] I cannot successfully save an image with an invalid URL.
- [ ] An error message is associated with the appropriate input field on
  failure.

__Discussion Topic__:
- [ ] What is a valid image URL?

__Reference material__:
- https://github.com/plataformatec/simple_form

__Dependencies__:
- Deployment Environment

__Note__: Many image sharing sites will permit one to upload images, or
download images when provided a link. For this project we will only work with
the link, i.e., URL, to various images.

__Note__: To understand the database tables created by the `acts_as_taggable` gem, you may find it useful to use a program like the [DB Browser for SQLite](http://sqlitebrowser.org/).

### PR DESCRIPTION
**Pull request by GithubofXzq**
Closes #2

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [ ] I have replaced `X` above with the associated issue number.
- [ ] I have changed the issue's label to "pull request".
- [ ] I have assigned the designated reviewer to this PR.
- [ ] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -47,3 +47,6 @@ end
 
 # Windows does not include zoneinfo files, so bundle the tzinfo-data gem
 gem 'tzinfo-data', platforms: %i[mingw mswin x64_mingw jruby]
+
+# add simple form
```
Comment (joshuasbates) [inline Gemfile]: This comment is unnecessary.  What do you think about removing it?

File: app/views/images/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<% if flash[:success].present? %>
```
Comment (joshuasbates) [inline app/views/images/show.html.erb]: I'd suggest moving this to the "app/views/layouts/application.html.erb" file as in the example in the article I sent you.  That way, the flash massage will be available on every page.  Does that make sense?

File: test/fixtures/images.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (joshuasbates) [inline test/fixtures/images.yml]: You can omit this fie from commit

File: app/views/images/show.html.erb (hunk 2 above)
Comment (joshuasbates) [reply app/views/images/show.html.erb]: Also note that a flash can have numerous types, like `error`, or `warning`... again, look at the example I provided which shows the right way to handle and display the flash message.

File: test/fixtures/images.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+one:
```
Comment (dshen94) [inline test/fixtures/images.yml]: Would these fixtures reflect what instances of your model is going to look like?

File: app/controllers/images_controller.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    @image = Image.find(params[:id])
+  end
+
+  def create
+    @image = Image.new(image_params)
+    if @image.save
+      flash[:success] = 'You have successfully added an image.'
+      redirect_to @image
+    else
+      render :new, status: :unprocessable_entity
```
Comment (dshen94) [inline app/controllers/images_controller.rb]: Do we want to add a failure flash message here since we had a flash message for the success case?

**Pull request merged**